### PR TITLE
[TTS] Ete tests

### PR DESCRIPTION
**Pull request by sudiksha27**
## General
**Before this PR**:
There was not ete testing for transactions 4

**After this PR**:
Adds a few ete tests for transactions 4.

**Priority**:
Early next week is ideal

**Concerns / possible downsides (what feedback would you like?)**:
Coverage, please point out any test cases that might be missing

**Is documentation needed?**:
No

## Compatibility
~**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:~

~**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:~

~**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:~

~**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:~

~**Does this PR need a schema migration?**~

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:
All tests

**What was existing testing like? What have you done to improve it?**:
Added tests

~**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:~

~**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:~

## Execution
~**How would I tell this PR works in production? (Metrics, logs, etc.)**:~

~**Has the safety of all log arguments been decided correctly?**:~

~**Will this change significantly affect our spending on metrics or logs?**:~

~**How would I tell that this PR does not work in production? (monitors, etc.)**:~

~**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:~

~**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:~

## Scale
~**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:~

~**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:~

~**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:~

## Development Process
**Where should we start reviewing?**:
`TransactionServicesTest`

~**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:~

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

**Comment by sudiksha27** [comment]
Blocked on https://github.com/palantir/atlasdb/pull/6326

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/service/TransactionServicesTest.java [hunk 1]
```diff
@@ -140,28 +148,133 @@ public void canCommitV2Transaction() {
...
+        forceInstallV4();
+        initializeTimestamps();
+        transactionService.putUnlessExists(startTs, commitTs);
+
+        assertThatThrownBy(() -> transactionService.getV2(startTs)).isInstanceOf(KeyAlreadyExistsException.class);
+    }
+
+    @Test
+    public void canCommitV4Transaction() {
+        forceInstallV4();
```
Comment (sudiksha27) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/service/TransactionServicesTest.java:208]: need tests for check and set

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/service/TransactionServicesTest.java [hunk 2]
```diff
@@ -140,28 +148,133 @@ public void canCommitV2Transaction() {
...
+        assertThat(actualMcasRequest.tableRef()).isEqualTo(TransactionConstants.TRANSACTIONS2_TABLE);
+        assertThat(ByteBuffer.wrap(actualMcasRequest.rowName())).isEqualTo(ByteBuffer.wrap(cell.getRowName()));
+
+        assertThat(actualMcasRequest.expected().keySet()).containsExactly(cell);
+        assertThat(actualMcasRequest.expected().get(cell)).containsExactly(TransactionConstants.TTS_IN_PROGRESS_MARKER);
+
+        assertThat(actualMcasRequest.updates().keySet()).containsExactly(cell);
+        assertThat(actualMcasRequest.updates().get(cell)).containsExactly(update);
+    }
+
```
Comment (sudiksha27) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/service/TransactionServicesTest.java:228]: also add tests for reading unknown.

**Comment by sudiksha27** [comment]
Blocked on https://github.com/palantir/atlasdb/pull/6357

File: atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionConstants.java [hunk 3]
```diff
@@ -47,6 +47,7 @@ private TransactionConstants() {
     public static final TransactionStatus ABORTED = TransactionStatuses.aborted();
     public static final TransactionStatus IN_PROGRESS = TransactionStatuses.inProgress();
     public static final TransactionStatus UNKNOWN = TransactionStatuses.unknown();
+    public static final byte[] TTS_IN_PROGRESS_MARKER = getTtsInProgressMarker();
```
Comment (Sam-Kramer) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionConstants.java:50]: Why not just do `byte[] TTS_IN_PROGRESS_MARKER = new byte[] {1};`?

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/service/TransactionServicesTest.java (hunk 1 above)
Comment (sudiksha27) [reply atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/service/TransactionServicesTest.java:212]: Oka that will have to be at the level of resilientAtomicTable

**Review by Sam-Kramer — APPROVED** [review]
Just the one nit

File: atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionConstants.java (hunk 3 above)
Comment (sudiksha27) [reply atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionConstants.java:50]: Yep I agree with that but checkstyle does not so much ` warning: [MutablePublicArray] Non-empty arrays are mutable` :(

**Pull request merged**